build-tools/packages/build-cli/src/commands/modify/fluid-imports.ts

```diff
@@ -0,0 +1,334 @@
+					const expectedLevel = getApiLevelForImportName(name, data, "public", onlyInternal);
+
+					log?.logIndent(
+						`Found import named: '${fullImportSpecifierText}' (${expectedLevel})`,
+						4,
+					);
+					const newSpecifier =
+						expectedLevel === "public" ? moduleName : `${moduleName}/${expectedLevel}`;
+
+					if (!newImports.has(newSpecifier)) {
+						newImports.set(newSpecifier, new Set());
+					}
+					newImports.get(newSpecifier)?.add(fullImportSpecifierText);
+				}
+			}
+		}
+
+		// Delete any header comments at the beginning of the file. Save the text so we can re-insert it at the end of
+		// processing. Note that this does modify the source file, but we only save changes if the imports are updated, so
+		// the removal will not be persisted unless there are import changes. In that case we re-add the header before we
+		// save. Therefore it's safe to remove the header here even before we know if we need to write the file.
+		const headerText = removeFileHeaderComment(sourceFile);
+
+		// Need to get declarations again because nodes are invalidated after calling replaceText above
+		imports = sourceFile.getImportDeclarations();
+
+		// Second pass: Update existing imports and add any missing ones
+		for (const importDeclaration of imports) {
+			// Skip non-Fluid imports
+			if (!isFluidImport(importDeclaration)) {
+				continue;
+			}
+
+			const [moduleName] = parseImport(importDeclaration);
+			const moduleSpecifier = importDeclaration.getModuleSpecifierValue();
+
+			// Skip Fluid imports that aren't in the data file
+			if (!mappingData.has(moduleName)) {
+				continue;
+			}
+
+			// Check if there are supposed to be any new imports from the module specifier
+			const newImportNames = newImports.get(moduleSpecifier);
+
+			// Since there are no imports from this module specifier, remove it.
+			if (newImportNames === undefined) {
+				importDeclaration.remove();
+				sourceFileChanged = true;
+				continue;
+			}
+
+			// There are new named imports for this module specifier, so remove all individual named imports and immediately
+			// add the new ones.
+			if (newImportNames.size > 0) {
+				importDeclaration.removeNamedImports();
+				importDeclaration.addNamedImports([...newImportNames]);
+				// Need to clear the list of new named imports since we just added them.
+				newImportNames.clear();
+				sourceFileChanged = true;
+			}
+		}
+
+		// Add any imports from a specifier that wasn't already in the file
+		for (const [importSpecifier, newImportNames] of newImports) {
+			if (newImportNames.size > 0) {
+				sourceFile.addImportDeclaration({
+					moduleSpecifier: importSpecifier,
+					namedImports: [...newImportNames],
+				});
+			}
+		}
+
+		if (sourceFileChanged) {
+			// Manually re-insert the header at the top of the file
+			sourceFile.insertText(0, headerText);
+
+			if (organizeImports) {
+				log?.info(`Organized imports in: ${sourceFile.getBaseName()}`);
+				sourceFile.organizeImports();
```

Comment:
Does this respect our sorting? Having trouble seeing utility if it doesn't. Maybe drop the option if it doesn't.?

Reply:
No, it doesn't use our sorting. It's equivalent to the default "organize imports" in vscode. I'm on the fence about its
utility. I think it could be helpful for non-Fluid code, but in that case it might be better to organize as a separate
step first, then rewrite the imports. And that points to this being a separate command or a mutually exclusive flag.

Reply:
Maybe note how it organizes in the help.